### PR TITLE
Endpoint and coop starring

### PR DESCRIPTION
**Pull request by Claire-Thib**
- implemented css for toggling between starred coops and non-starred coops
- stars are rendered correctly
- implementation of toggling starred attribute of coop does not work yet

Our Error:
- line 308 of FiltersPage, we pass in a function toggleStar as a prop. Profile.js is reading that prop as null and FiltersPage is calling toggleStar on its own

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/usr/bin/python3"
+}
```
Comment (rrrliu) [inline .vscode/settings.json]: don't check in changes in `.vscode/`

File: client/src/components/Card/Card.js [hunk 2]
```diff
@@ -9,13 +9,37 @@ export default function CardComponent(props) {
...
+    if (props.starred) {
+      return (
+        <svg className="card-starred" viewBox="0 0 20 20">
+          <path fill='#ffc011' d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"> </path>
+        </svg>
+      );
+    } else {
+      return (
+        <svg className="card-not-starred" viewBox="0 0 20 20">
+          <path d="M17.684,7.925l-5.131-0.67L10.329,2.57c-0.131-0.275-0.527-0.275-0.658,0L7.447,7.255l-5.131,0.67C2.014,7.964,1.892,8.333,2.113,8.54l3.76,3.568L4.924,17.21c-0.056,0.297,0.261,0.525,0.533,0.379L10,15.109l4.543,2.479c0.273,0.153,0.587-0.089,0.533-0.379l-0.949-5.103l3.76-3.568C18.108,8.333,17.986,7.964,17.684,7.925 M13.481,11.723c-0.089,0.083-0.129,0.205-0.105,0.324l0.848,4.547l-4.047-2.208c-0.055-0.03-0.116-0.045-0.176-0.045s-0.122,0.015-0.176,0.045l-4.047,2.208l0.847-4.547c0.023-0.119-0.016-0.241-0.105-0.324L3.162,8.54L7.74,7.941c0.124-0.016,0.229-0.093,0.282-0.203L10,3.568l1.978,4.17c0.053,0.11,0.158,0.187,0.282,0.203l4.578,0.598L13.481,11.723z"></path>
```
Comment (rrrliu) [inline client/src/components/Card/Card.js]: same here

File: client/src/components/Profile/Profile.js [hunk 3]
```diff
@@ -260,12 +274,31 @@ export default function Profile(props) {
...
+  function renderStar() {
+    if (starred) {
+      return (
+        <svg className="profile-starred" viewBox="0 0 20 20" onClick={() => handleStar(coopId, starrerId)}>
+          <path fill='#ffc011' d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"></path>
+        </svg>
+      );
+    } else {
+      return (
+        <svg className="profile-not-starred" viewBox="0 0 20 20" onClick={() => handleStar(coopId, starrerId)}>
```
Comment (leebianca) [inline client/src/components/Profile/Profile.js]: use `props.handleStar` instead. we don't need to be setting `props.handleStar` as a state in order to use it

File: client/src/pages/FiltersPage/FiltersPage.js [hunk 4]
```diff
@@ -15,171 +9,93 @@ import React from 'react';
...
+  if (!coops) {
+    return <div>Loading...</div>;
+  }
+
+  
+  function toggleStar(starredId, starrerId) {
+    if (starredCoops.includes(starredId)) {
+      //if the coop is already starred
+      //remove the coop from the list of starred
+      console.log("HERE")
```
Comment (leebianca) [inline client/src/pages/FiltersPage/FiltersPage.js]: remove

File: client/src/pages/FiltersPage/FiltersPage.js [hunk 5]
```diff
@@ -188,11 +104,30 @@ export default function FiltersPage() {
...
+            tags={coop.tags}
+            starred={starredCoops.includes(coop.id)}
+            selected={selectedIndex === index}
+            onClick={() => renderProfile(coop, index)}
+          />
+        ))}
+      </div>
+    );
+  } */
+
```
Comment (leebianca) [inline client/src/pages/FiltersPage/FiltersPage.js]: remove

File: client/src/pages/FiltersPage/ReactToggle.css [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  border-radius: 30px;
+  background-color: #81a1b7;
+  -webkit-transition: all 0.2s ease;
+  -moz-transition: all 0.2s ease;
+  transition: all 0.2s ease;
+}
+
+/* .react-toggle:hover:not(.react-toggle--disabled) .react-toggle-track {
+  background-color: #d1dbe2;
+} */
```
Comment (leebianca) [inline client/src/pages/FiltersPage/ReactToggle.css]: remove

File: client/src/components/Profile/Profile.js [hunk 7]
```diff
@@ -260,12 +274,31 @@ export default function Profile(props) {
...
+
   function renderContact() {
     return (
       <div className="profile-pic-text-container">
         <img className="profile-pic" alt="Image" src={profilePicture} />
         <div className="profile-text-container">
-          <b className="profile-co-op-name">{name}</b>
+          <div className="card-name-star-wrapper">
+            <b className="profile-co-op-name">{name}</b>
+            {renderStar()}
```
Comment (leebianca) [inline client/src/components/Profile/Profile.js:290]: it seems like the star is stretching out `profile-text-container` and making the contact links clickable way to the right. adding `width: fit-content` to `profile-co-op-contact-link` should fix it

File: client/src/components/Profile/Profile.js [hunk 8]
```diff
@@ -18,23 +20,35 @@ export default function Profile(props) {
...
+    setInstaLink(props.coop.insta_link);
     setEmail(props.coop.email);
     setWebsite(props.coop.website);
-    setProfilePicture(props.coop.profilePicture);
+    setProfilePicture(props.coop.profile_pic);
+    //set the starred to the coop
+    setStarred(props.starred);
+    //set the handle star method
+    setHandleStar(props.handleStar);
+    console.log("HANDLE STAR: " + props.h)
```
Comment (leebianca) [inline client/src/components/Profile/Profile.js]: remove

File: db/cofed.sql [hunk 9]
```diff
@@ -47,3 +51,25 @@ GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA public TO root;
...
+
+INSERT INTO stars
+    (starred_coop_id, starrer_coop_id)
+VALUES
+    (1, 2),
+    (1, 3),
+    (2, 1),
+    (4, 1),
+    (5, 2),
+    (6, 1);
```
Comment (rrrliu) [inline db/cofed.sql]: newline at end of file

File: db/index.js [hunk 10]
```diff
@@ -3,4 +3,6 @@ const pool = new Pool({ database: 'cofed' });
 
 require('dotenv').config();
 
+// const pool = new Pool();
+
```
Comment (rrrliu) [inline db/index.js]: remove

File: client/src/pages/FiltersPage/FiltersPage.js (hunk 5 above)
Comment (rrrliu) [inline client/src/pages/FiltersPage/FiltersPage.js]: remove

File: client/src/pages/FiltersPage/FiltersPage.js (hunk 4 above)
Comment (rrrliu) [inline client/src/pages/FiltersPage/FiltersPage.js]: remove

File: client/src/components/Profile/Profile.js [hunk 11]
```diff
@@ -18,23 +20,35 @@ export default function Profile(props) {
...
+    setFbLink(props.coop.fb_link);
+    setInstaLink(props.coop.insta_link);
     setEmail(props.coop.email);
     setWebsite(props.coop.website);
-    setProfilePicture(props.coop.profilePicture);
+    setProfilePicture(props.coop.profile_pic);
+    //set the starred to the coop
+    setStarred(props.starred);
+    //set the handle star method
+    setHandleStar(props.handleStar);
```
Comment (rrrliu) [inline client/src/components/Profile/Profile.js]: if these are never changed within the Profile component, they should just remain props. don't create a state for them

File: api/routes/index.js [hunk 12]
```diff
@@ -14,4 +14,69 @@ router.get('/', async (req, res) => {
...
+router.delete('/delete', async (req, res) => {
+  const { starredId } = req.params;
+  const { starrerId } = req.params;
+  try {
+    const query = await db.query(
+      //DONE: delete row from stars table
+      `DELETE FROM stars 
+      WHERE starred_coop_id = $1 
+      AND starrer_coop_id = $2;`,
+      [request.body.starredId, request.body.starrerId]
```
Comment (rrrliu) [inline api/routes/index.js]: `request` doesn't exist in this context (check the parameters of this callback) -- you should be able to use `starrerId` and `starredId` that you declared above

File: client/src/components/Card/Card.js [hunk 13]
```diff
@@ -9,13 +9,37 @@ export default function CardComponent(props) {
     'card-selected': props.selected,
   });
 
+
+  function renderStar() {
+    if (props.starred) {
+      return (
+        <svg className="card-starred" viewBox="0 0 20 20">
+          <path fill='#ffc011' d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"> </path>
```
Comment (rrrliu) [inline client/src/components/Card/Card.js]: move to its own .svg file

**Review by leebianca — CHANGES_REQUESTED** [review]
you guys 🥺🥺 y'all have come so far ❤️ great job on this task this week! ik it wasn't easy

i left a few comments, and to address your issue with `toggleStar`, you're setting it as a state when it doesn't need to be set as a state. if y'all get rid of the `handleStar` state and fix the comment about `props.handleStar`, that should fix it. on a higher level though, since "starring" is a feature on the profile page, shouldn't the `toggleStar` function be contained within `Profile.js`? what were y'alls reasoning for putting it in `FiltersPage`?

otherwise, it's looking good so far!!

File: api/routes/index.js [hunk 14]
```diff
@@ -14,4 +14,69 @@ router.get('/', async (req, res) => {
...
+  } catch (error) {
+    console.log(error.stack);
+  }
+});
+
+//post new starred
+router.post('/addStar', async (request, response) => {
+  try {
+    const query = await db.query(
+      `INSERT INTO stars (starred_coop_id, starrer_coop_id))
```
Comment (rrrliu) [inline api/routes/index.js]: sql syntax error here (check the parentheses)

File: client/src/pages/FiltersPage/FiltersPage.js [hunk 15]
```diff
@@ -15,171 +9,93 @@ import React from 'react';
...
+    } else {
+      //if the coop isn't starred yet
+      //add the coop from the list of starred
+      starredCoops.push(starredId);
+      //make a post request
+      axios.post('/api/addStar', {
+        data: {
+          starred_id: starredId,
+          starrer_id: starrerId,
+        },
```
Comment (rrrliu) [inline client/src/pages/FiltersPage/FiltersPage.js]: again, if you're using a an additional data key, make sure you properly handle that in your router.post

File: client/src/pages/FiltersPage/FiltersPage.js [hunk 16]
```diff
@@ -15,171 +9,93 @@ import React from 'react';
...
+      const index = starredCoops.indexOf(starredId);
+      if (index > -1) {
+        starredCoops.splice(index, 1);
+      }
+      //remove the row from the database
+      axios.delete('/api/delete', {
+        data: {
+          starred_id: starredId,
+          starrer_id: starrerId,
+        },
```
Comment (rrrliu) [inline client/src/pages/FiltersPage/FiltersPage.js]: this body format doesn't match what your router.delete handler expects, which is another reason it's not working. why is there a `data` key, and make sure the keys match. it looks like you're putting `starred_coop_id` here but your router.delete handler is expecting `starredId`

**Review by rrrliu — CHANGES_REQUESTED** [review]
gjgj 🤩 !!! look at my comments for advice on how to debug the starring/unstarring API request

also make sure to run `yarn run lint`

**Comment by ilyues** [comment]
> since "starring" is a feature on the profile page, shouldn't the `toggleStar` function be contained within `Profile.js`? what were y'alls reasoning for putting it in `FiltersPage`?

@leebianca toggleStar changes the array of starred coop ids & both Card.js and Profile.js take in props that use this array - is there a way to change this array from inside Profile.js without passing in a function to change it?

**Comment by leebianca** [comment]
> > since "starring" is a feature on the profile page, shouldn't the `toggleStar` function be contained within `Profile.js`? what were y'alls reasoning for putting it in `FiltersPage`?
> 
> @leebianca toggleStar changes the array of starred coop ids & both Card.js and Profile.js take in props that use this array - is there a way to change this array from inside Profile.js without passing in a function to change it?

oh yea, in that case, you could use [callback functions](https://webomnizz.com/change-parent-component-state-from-child-using-hooks-in-react/) to change states in the parent component from the child component. just questioning the design decision though :) feel free to just leave it there if it's easier

File: client/src/assets/starred.svg [hunk 17]
```diff
@@ -0,0 +1,4 @@
+<svg viewBox="0 0 20 20" fill="#ffc011" xmlns="http://www.w3.org/2000/svg">
+  <path d="M9.5 14.25l-5.584 2.936 1.066-6.218L.465 6.564l6.243-.907L9.5 0l2.792 5.657 6.243.907-4.517 4.404 1.066 6.218"
+  ></path>
+</svg>
```
Comment (rrrliu) [inline client/src/assets/starred.svg]: add newline

File: client/src/assets/unstarred.svg [hunk 18]
```diff
@@ -0,0 +1,4 @@
+<svg viewBox="0 0 20 20" fill="#81a1b7" xmlns="http://www.w3.org/2000/svg">
+  <path d="M17.684,7.925l-5.131-0.67L10.329,2.57c-0.131-0.275-0.527-0.275-0.658,0L7.447,7.255l-5.131,0.67C2.014,7.964,1.892,8.333,2.113,8.54l3.76,3.568L4.924,17.21c-0.056,0.297,0.261,0.525,0.533,0.379L10,15.109l4.543,2.479c0.273,0.153,0.587-0.089,0.533-0.379l-0.949-5.103l3.76-3.568C18.108,8.333,17.986,7.964,17.684,7.925 M13.481,11.723c-0.089,0.083-0.129,0.205-0.105,0.324l0.848,4.547l-4.047-2.208c-0.055-0.03-0.116-0.045-0.176-0.045s-0.122,0.015-0.176,0.045l-4.047,2.208l0.847-4.547c0.023-0.119-0.016-0.241-0.105-0.324L3.162,8.54L7.74,7.941c0.124-0.016,0.229-0.093,0.282-0.203L10,3.568l1.978,4.17c0.053,0.11,0.158,0.187,0.282,0.203l4.578,0.598L13.481,11.723z"
+  ></path>
+</svg>
```
Comment (rrrliu) [inline client/src/assets/unstarred.svg]: add newline

File: db/index.js (hunk 10 above)
Comment (rrrliu) [reply db/index.js]: remove

**Review by rrrliu — DISMISSED** [review]

**Review by leebianca — DISMISSED** [review]

**Pull request closed without merging**